kotlin-native/backend.native/tests/objcexport/coroutines.swift

```diff
@@ -48,6 +48,36 @@ private func testUnitCallSimple() throws {
     try assertNil(error)
 }
 
+private func testUnitCallNonMainDispatcher() throws {
+#if LEGACY_SUSPEND_UNIT_FUNCTION_EXPORT
+    var result: KotlinUnit? = nil
+#endif
+    var error: Error? = nil
+    var completionCalled = 0
+
+#if ALLOW_SUSPEND_ANY_THREAD
+#if LEGACY_SUSPEND_UNIT_FUNCTION_EXPORT
+    DispatchQueue.global().async {
+        CoroutinesKt.unitSuspendFun { _result, _error in
+            completionCalled += 1
+            result = _result
+            error = _error
+        }
+    }
+#else
+    DispatchQueue.global().async {
+        CoroutinesKt.unitSuspendFun { _error in
+            completionCalled += 1
+            error = _error
+        }
+    }
+#endif
+#endif
+
+    try assertNil(error)
```

Comment:
You'd need to ensure that the background queue has finished executing blocks by the time this is asserted.

Reply:
done.